azurerm/resource_arm_storage_account_blob_settings.go

```diff
@@ -0,0 +1,177 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2019-04-01/storage"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+var storageAccountBlobSettingsResourceName = "azurerm_storage_account_blob_settings"
+
+func resourceArmStorageAccountBlobSettings() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmStorageAccountBlobSettingsCreateUpdate,
+		Read:   resourceArmStorageAccountBlobSettingsRead,
+		Update: resourceArmStorageAccountBlobSettingsCreateUpdate,
+		Delete: resourceArmStorageAccountBlobSettingsDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"storage_account_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"enable_soft_delete": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"soft_delete_retention_days": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      7,
+				ValidateFunc: validation.IntBetween(1, 365),
+			},
+		},
+	}
+}
+
+func resourceArmStorageAccountBlobSettingsCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).storageBlobServicesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure ARM Storage Account Blob Settings creation.")
+
+	resourceGroupName := d.Get("resource_group_name").(string)
+	storageAccountName := d.Get("storage_account_name").(string)
+	enableSoftDelete := d.Get("enable_soft_delete").(bool)
+	softDeleteRetentionDays := int32(d.Get("soft_delete_retention_days").(int))
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.GetServiceProperties(ctx, resourceGroupName, storageAccountName)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("error checking for presence of existing Storage Account Blob Settings (Storage Account %q / Resource Group %q): %s",
+					storageAccountName, resourceGroupName, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError(storageAccountBlobSettingsResourceName, *existing.ID)
+		}
+	}
```

Comment:
since the Storage Account will always have properties, this won't work/we'd need to remove this